File: frontend/src/landing/app/types/match.ts
export interface ObjectivesSummary {
  readonly baron: number;
  readonly dragon: number;
  readonly rift_herald: number;
  readonly tower: number;
  readonly inhibitor: number;
}

export interface ParticipantDetail {
  readonly puuid: string;
  readonly riot_id: string | null;
  readonly champion_id: number;
  readonly champion_name: string;
  readonly position: string;
  readonly win: boolean;
  readonly kills: number;
  readonly deaths: number;
  readonly assists: number;
  readonly cs: number;
  readonly gold_earned: number;
  readonly damage_to_champions: number;
  readonly vision_score: number;
  readonly items: readonly number[];
  readonly summoner_spells: readonly number[];
  readonly is_viewer: boolean;
}

export interface TeamDetail {
  readonly team_id: number;
  readonly win: boolean;
  readonly bans: readonly number[];
  readonly objectives: ObjectivesSummary;
  readonly participants: readonly ParticipantDetail[];
}

export interface MatchDetail {
  readonly match_id: string;
  readonly game_creation: number;
  readonly game_duration: number;
  readonly game_version: string;
  readonly queue_id: number;
  readonly queue_label: string;
  readonly map_id: number;
  readonly map_label: string;
  readonly teams: readonly TeamDetail[];
}

File: backend/app/schemas/match.py
from pydantic import BaseModel


class ObjectivesSummaryResponse(BaseModel):
    baron: int
    dragon: int
    rift_herald: int
    tower: int
    inhibitor: int


class ChampionBanResponse(BaseModel):
    champion_id: int
    champion_name: str


class ParticipantDetailResponse(BaseModel):
    puuid: str
    riot_id: str | None
    champion_id: int
    champion_name: str
    position: str
    win: bool
    kills: int
    deaths: int
    assists: int
    cs: int
    gold_earned: int
    damage_to_champions: int
    vision_score: int
    items: list[int]
    summoner_spells: list[int]
    is_viewer: bool


class TeamDetailResponse(BaseModel):
    team_id: int
    win: bool
    bans: list[ChampionBanResponse]
    objectives: ObjectivesSummaryResponse
    participants: list[ParticipantDetailResponse]


class MatchDetailResponse(BaseModel):
    match_id: str
    game_creation: int
    game_duration: int
    game_version: str
    queue_id: int
    queue_label: str
    map_id: int
    map_label: str
    teams: list[TeamDetailResponse]


class MatchHistorySummaryResponse(BaseModel):
    match_id: str
    champion_name: str
    outcome: str
    duration_minutes: int
    map_label: str
    played_on: str
    kills: int
    deaths: int
    assists: int
    cs: int
    position: str
